lib/netsuite/support/search_result.rb

```diff
@@ -59,11 +59,22 @@ def initialize(response, result_class, credentials)
                 # skip all attributes: look for :basic and all :xxx_join
                 next if search_group.to_s.start_with?('@')
 
-                record[search_group].each_pair do |k, v|
+                # avoids `RuntimeError: can't add a new key into hash during iteration`
+                record[search_group][:custom_field_list] ||= {custom_field: []}
+
+                record[search_group].each_pair do |attr_name, search_result|
+                  # example pair:
+                  # {
+                  #   :department=>{
+                  #     :search_value=>{:@internal_id=>"113"},
```

Comment:
`search_value` is the value of the search result, right? Not the input into the origin search query?